code-style/javascript/README.md

```diff
@@ -0,0 +1,64 @@
+JavaScript
+====
+
+[![JS Logo][producti]][product]
+
+We write a lot of JavaScript. As the Web continues to grow and as users expect more dynamic interaction with websites, JavaScript has taken on an increasingly large role. However, we believe that it's also important to know when _not_ to use JavaScript. We prefer semantic HTML and CSS solutions whenever possible, and only reach for JavaScript to enhance these capabilities. We find that this approach helps to ensure that our work is accessible to the largest number of people possible.
+
+Style
+-----
+
+Consistent code styling helps our teams work together better and can help prevent common JavaScript pitfalls. We rely heavily on manual code reviews to ensure the quality and functionality of the code we're writing, but prefer to automate code style enforcement as much as possible.
+We use [ESLint][eslint] with a [custom configuration][eslint_config] to help enforce consistent code style.
+
+Here are a few guidelines we like to follow:
+
+- Prefer [ES6][es6] syntax, including
+  - [import][import]
+  - [spread operator][spread]
```

Comment:
Looking for other guidelines to add here

Reply:
I think I've seen two opinions on spread:
1. it's great, use it everywhere
2. it's great, but quickly becomes hard to follow. Use with caution.

How do we express that? Or am I misremembering?

Reply:
This is a good point, I think that the biggest downside of using spread is potential performance implications.

https://medium.com/@eghbali/what-is-the-es6-spread-operator-and-why-you-shouldnt-use-it-57c056078ed9

This touches on a much bigger question: Should we prefer optimizing our code for performance, or readability/maintainability? I tend to prefer maintainability until performance becomes an issue, the exact opposite conclusion the author of the above article comes to 😂. But I don't want my opinion to speak for all of Sparkbox, so I think I'll solicit other opinions in dev.